lua/plenary/path.lua

```diff
@@ -917,14 +921,20 @@ end
 
 function Path:find_upwards(filename)
   local folder = Path:new(self)
-  while self:absolute() ~= path.root do
+  local root = path.root(folder:absolute())
+
+  while true do
     local p = folder:joinpath(filename)
     if p:exists() then
       return p
     end
+    if folder:absolute() == root then
+      break
+    end
     folder = folder:parent()
   end
-  return ""
+
+  return nil
```

Comment:
While I agree with the `nil` change here, it do break backwards compatibility for existing code, maybe we should avoid that?

Reply:
So, I don't think it was actually even possible to reach this line before. It either returned at the line above, or was trapped in an infinite loop.

I added some specs and tried them out on the code before this PR and it gets stuck when it's unable to find the path.

Reply:
Ah yes, you are totally right :+1: